language/stdlib/modules/offer.mvir

```diff
@@ -0,0 +1,40 @@
+// TODO: add optional timeout for reclaiming by original publisher once we have implemented time
+module Offer {
+    // A wrapper around value `offered` that can be claimed by the address stored in `for`.
+    resource T<Offered> { offered: Offered, for: address }
+
+    // Publish a value of type `Offered` under the sender's account. The value can be claimed by
+    // either the `for` address or the transaction sender.
+    public create<Offered>(offered: Offered, for: address) {
+        move_to_sender<T<Offered>>(T<Offered> { offered: move(offered), for: move(for) });
+        return;
+    }
+
+    // Claim the value of type `Offered` published at `offer_address`.
+    // Only succeeds if the sender is the intended recipient stored in `for` or the original
+    // publisher `offer_address`.
+    // Also fails if no such value exists.
+    public redeem<Offered>(offer_address: address): Offered acquires T {
+        let offered: Offered;
+        let for: address;
+
+        T<Offered> { offered, for } = move_from<T<Offered>>(copy(offer_address));
+        assert(
+            get_txn_sender() == move(for) || get_txn_sender() == move(offer_address),
```

Comment:
I'm wondering if this can be redeemed by accounts who have already given their `WithdrawalCapability` to other accounts. Is this possible? or WithdrawalCapability is completely irrelevant to resource publishing (or redeeming in our case)?

Reply:
Good question. Yes, you can still claim for address `A` if you've delegated your `WithdrawalCapability` elsewhere. All that's required is the ability to send a transaction from `A`. As you say, `WithdrawalCapability` is irrelevant to publishing/redeeming.